tf-build-tools/tf-maven-plugin/src/main/java/com/deliveredtechnologies/maven/terraform/TerraformUpload.java

```diff
@@ -0,0 +1,95 @@
+package com.deliveredtechnologies.maven.terraform;
+
+
+import com.deliveredtechnologies.io.CommandLine;
+import com.deliveredtechnologies.io.Executable;
+import com.deliveredtechnologies.terraform.TerraformException;
+import com.deliveredtechnologies.terraform.TerraformUtils;
+import com.deliveredtechnologies.terraform.api.TerraformOperation;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+
+/**
+ * upload plan files to s3 after tf:plan.
+ */
+public class TerraformUpload implements TerraformOperation<String> {
```

Comment:
Why is this TerraformUpload if it only uploads to S3? Why not TerraformS3Upload?

Think about this: What happens if we want to upload the plan somewhere else? This class should not have to change.